staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt-config.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filters
+
+import (
+	"math"
+	"sort"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	cache "k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	rmtypesv1a1 "k8s.io/api/flowcontrol/v1alpha1"
+)
+
+// initialSync does the initial setup of the configuration of the
+// reqmgmt filter.  It either succeeds and returns `true` or logs an
+// error and returns `false.
+func (reqMgmt *requestManagement) initialSync() bool {
+	rms := &RMState{
+		flowSchemas:         make(FlowSchemaSeq, 0),
+		priorityLevelStates: make(map[string]*PriorityLevelState),
+	}
+	reqMgmt.curState.Store(rms)
+	reqMgmt.plInformer.AddEventHandler(reqMgmt)
+	reqMgmt.fsInformer.AddEventHandler(reqMgmt)
+	stopCh := make(chan struct{})
```

Comment:
leave a TODO here, plumbing stop channel from server.

Reply:
Yeah, there is really a major plumbing TODO, and the stop channel is just part of it, the shared informer factory is another part.